types/underscore/index.d.ts

```diff
@@ -96,7 +96,7 @@ declare module _ {
         (value: T): boolean;
     }
 
-    interface CollectionIterator<T extends TypeOfCollection<V, any>, TResult, V = Collection<T>> {
```

Comment:
As a point of interest, TypeOfCollection was actually a conditional type at one point, but I changed it to being a union because otherwise using TypeOfList or TypeOfDictionary as type `T` surprisingly ended up failing this constraint. Happily switching the constraint to the former type union for TypeOfCollection ended up being a viable workaround here 😄